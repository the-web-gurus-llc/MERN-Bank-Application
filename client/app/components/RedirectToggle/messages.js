/*
 * RedirectToggle Messages
 *
 * This contains all the text for the RedirectToggle component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.RedirectToggle';

export default defineMessages({
  register: {
    id: `${scope}.register`,
    defaultMessage: 'Registration',
  },
  login: {
    id: `${scope}.login`,
    defaultMessage: 'Log In',
  },
  registerContent: {
    id: `${scope}.registerContent`,
    defaultMessage:
      'If you do not have an account in our bank, you can create now by clicking on',
  },
  loginContent: {
    id: `${scope}.loginContent`,
    defaultMessage:
      'If you already have a bank account, you can log in by clicking on',
  },
});
