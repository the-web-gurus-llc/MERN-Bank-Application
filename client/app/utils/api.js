const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
