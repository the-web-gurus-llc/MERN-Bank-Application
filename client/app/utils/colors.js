export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
