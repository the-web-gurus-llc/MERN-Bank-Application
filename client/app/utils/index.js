export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
