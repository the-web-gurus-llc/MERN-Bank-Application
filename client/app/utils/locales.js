exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
