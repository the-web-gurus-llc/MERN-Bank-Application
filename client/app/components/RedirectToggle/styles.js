import styled from 'styled-components';
import { Button } from 'antd';
import { typography, colors } from 'utils';

export const StyledRedirectToggle = styled.div`
  margin: 20px auto 30px;
  text-align: left;
  font-weight: ${typography.fontWeightBold};

  padding: 0 15px;
  max-width: 560px;
`;

export const StyledButton = styled(Button)`
  font-weight: ${typography.fontWeightBold};
  color: ${colors.primaryBlue};
  padding: 0;
  height: auto;
`;
