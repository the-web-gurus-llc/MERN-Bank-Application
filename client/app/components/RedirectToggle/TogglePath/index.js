import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { push } from 'connected-react-router';
import { StyledButton } from '../styles';

export default function TogglePath({ name, path }) {
  const dispatch = useDispatch();
  const onChangeRoute = (route) => dispatch(push(route));

  return (
    <StyledButton type="link" onClick={() => onChangeRoute(path)}>
      {name}
    </StyledButton>
  );
}

TogglePath.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
};
