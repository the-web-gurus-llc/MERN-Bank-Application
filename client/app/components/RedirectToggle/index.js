/**
 *
 * RedirectToggle
 *
 */

import React from 'react';
import { useSelector } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage, intlShape, injectIntl } from 'react-intl';
import { routes } from 'utils';
import { makeSelectLocation } from 'containers/App/selectors';
import { StyledRedirectToggle } from './styles';
import TogglePath from './TogglePath';
import messages from './messages';

const stateSelector = createStructuredSelector({
  location: makeSelectLocation(),
});

function RedirectToggle({ intl }) {
  const {
    location: { pathname },
  } = useSelector(stateSelector);

  return (
    <StyledRedirectToggle>
      {pathname === routes.login.path ? (
        <>
          <FormattedMessage {...messages.registerContent} />{' '}
          <TogglePath
            name={intl.formatMessage(messages.register)}
            path={routes.register.path}
          />
        </>
      ) : (
        <>
          <FormattedMessage {...messages.loginContent} />{' '}
          <TogglePath
            name={intl.formatMessage(messages.login)}
            path={routes.login.path}
          />
        </>
      )}
    </StyledRedirectToggle>
  );
}

RedirectToggle.propTypes = {
  intl: intlShape.isRequired,
};

export default injectIntl(RedirectToggle);
