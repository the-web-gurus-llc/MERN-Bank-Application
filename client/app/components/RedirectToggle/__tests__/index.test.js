import React from 'react';
import { render } from '@testing-library/react';

import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import configureStore from 'utils/configureStore';
import RedirectToggle from '../index';

describe('<RedirectToggle />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a information', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <RedirectToggle />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
