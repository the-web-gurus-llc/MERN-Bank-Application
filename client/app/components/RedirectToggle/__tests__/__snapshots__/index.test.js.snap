// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RedirectToggle /> should render a information 1`] = `
<div
  class="RedirectTogglestyle__StyledRedirectToggle-sc-15lqog-0 WiacX"
>
  <span>
    If you already have a bank account, you can log in by clicking on
  </span>
   
  <button
    class="ant-btn RedirectTogglestyle__StyledButton-sc-15lqog-1 cEWUtu ant-btn-link"
    type="button"
  >
    <span>
      Log In
    </span>
  </button>
</div>
`;
